### DISCUSSION TITLE
How to connect to a MariaDB server on MacOS

### DISCUSSION THREAD

**Discussion by madalinignisca** [chris2511/xca#613] [Q&A]
Hi. I'm on latest MacOS 15.1.1, using XCA 2.8.0. Local it works fine.

I thought to centralise our database with a few colleagues, and wanted to move it to a MariaDB server (privately accessible).

I've open the remote database dialog window and the only option is ODBC. What should I fill in the DSN field to make it work?

<img width="814" alt="image" src="https://github.com/user-attachments/assets/3bf2a387-d22f-44ff-81c0-babcb8f5db5b">

**Comment by chris2511** [comment]
I screwed it up:
> $  otool -L xca-2.8.0.app/Contents/PlugIns/sqldrivers/libqsqlmysql.dylib
> xca-2.8.0.app/Contents/PlugIns/sqldrivers/libqsqlmysql.dylib (architecture arm64):
	/Users/chris/build/xca-dir/install/lib/libmariadb.3.dylib (compatibility version 3.0.0, current version 3.0.0)

The plugin references the mariadb  library by an absolute path instead of a relative one.
Changing the name inside the library with:
>  install_name_tool -change /Users/chris/build/xca-dir/install/lib/libmariadb.3.dylib @rpath/libmariadb.3.dylib xca-2.8.0.app/Contents/PlugIns/sqldrivers/libqsqlmysql.dylib

invalidates the signature.

**Reply by FOS-Networks** [reply]
Will a new version be released soon to fix this issue?

**Reply by vdovhanych** [reply]
I tried fixing the path, but yeah, I can't launch the app since it complaints about the app being broken and not even removing all attributes allows it to be launched.

**Reply by marek-liska** [reply]
you can use `codesign --force --deep --sign - /Applications/xca.app`
